### DISCUSSION TITLE
Error / Request when opening VS Code IBM i extension

### DISCUSSION THREAD

**Discussion by Gregory1701** [codefori/vscode-ibmi#1266] [IBM i]
When I have a source member open in the IBM i extension, then close VS Code - leaving the source member open.  The next time I open VS Code I get an error message:
![image](https://user-images.githubusercontent.com/132368707/235687416-9a9ddce2-4d75-4dc3-98a2-c92a1752d8e7.png)
I believe this is due to it not automatically connecting to the last IBM i I was connected to.  I'm not able to find a setting where I can tell it to 'autoconnect' to an iSeries upon startup.  Can someone point me in the right direction?  Or can this be added?

Thanks,
Gregory

**Comment by worksofliam** [comment]
As of today, there is no 'auto connect' functionality and right now I don't think we have plans to add one.

@chrjorgensen as the person who came up with Quick Connect, I'd be curious as to what you think about this?

**Reply by sebjulliand** [reply]
Just auto-connecting would not solve the problem as vscode queries the QSYS file system right after it started. So the error would still show up as the error states it cannot find the provider for the `member://` fs.

However, I did a quick test an managed to solve the issue by:
- Adding `onFileSystem:member` to the extension's `activationEvents`
- Having the QSYS file system to invoke the `code-for-ibmi.connectToPrevious` command when `readFile` is called and no connection is found.

Here is how it looks - i opened a member, then reload VSCode. You can see it will reconnect before displaying the member, with no issue:
![fs_reconnect](https://user-images.githubusercontent.com/11096890/235713330-150fb0a5-a6f3-42ea-8a56-7fdb7f68ef6e.gif)

@worksofliam, @chrjorgensen I can push a PR if you want; I've got it working for both the QSYS and IFS file systems 😊

**Reply by chrjorgensen** [reply]
@sebjulliand Go! You are a superhero!

**Reply by sebjulliand** [reply]
Be my guest then 😁 https://github.com/halcyon-tech/vscode-ibmi/pull/1268

**Reply by Gregory1701** [reply]
Thanks sebjulliand! In your video, are you having VS Code automatically view the IBM i extension upon opening? If so, how can I do that?

**Reply by worksofliam** [reply]
@Gregory1701 Right now it's in a PR (pull request) and it will likely be out later this week or next week.

**Reply by Gregory1701** [reply]
Awesome, thanks!

**Comment by Gregory1701** [comment]
This is an old issue which was solved long ago. Thanks all. Closing.
